versions/development/SPEC.md

```diff
@@ -698,6 +699,40 @@ Pair values can also be specified within the [workflow inputs JSON](https://gith
 }
 ```
 
+### Optional literals
+
+Any non-optional value can be wrapped into an optional value by assigning it to an optional variable:
+
+```WDL
+Int? maybe_integer = 5
+```
+
+The `None` literal is the value that an optional has when it is not defined.
+
+```WDL
+Int? maybe_integer = None
+# maybe_integer is not defined
+```
+
+This is useful in a number of cases, for example when an output of a command depends on a certain flag:
+```wdl
+task test
+  input {
+    File inputFile
+    String outputPath
+    Boolean log
+  }
+  command {
+    test ~{true="--log" false="" log} inputFile outputPath
+  }
+  output {
+    File? logFile = if log then "test.log" else None
+  }
+}
+```
+If `log` is true, the command will evaluate to `test --log inputFile outputPath` and `logFile` will be a defined optional.
```

Comment:
Given that `None` is _the_ undefined optional, the wording of the phrase "will be a defined optional" seems a bit awkward. You could just as easily say "will be `None`". Here you could instead say:

When `log` is `true`, the output `logFile` has the value `"test.log"`. If it's `false`, it has the value `None`.